### PR TITLE
Add pre-commit config yaml file to enable automatic code formatting

### PR DESCRIPTION
**Pull request by polinaeterna**
@huggingface/datasets do you think it would be useful? Motivation - sometimes PRs are like 30% "fix: style" commits :)

If so - I need to double check the config but for me locally it works as expected.

### CODE REVIEW

**Review by lhoestq — APPROVED** [review]
sounds good to me - maybe @albertvillanova is also familiar with this ?

**Review by nateraw — APPROVED** [review]
LGTM! 🤗

Only gripe I have with pre-commit is when you're forced to use it. With this being optional, its a super nice addition for folks who prefer to use it vs Makefile :). Thank you for adding

**Comment by Skylion007** [comment]
Better yet have someone enable pre-commit CI https://pre-commit.ci/ and it will apply the pre-commit fixes to the PR automatically as an additional commit.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+repos:
```
[the remark this answers is not in the document]
Comment (Skylion007) [reply .pre-commit-config.yaml:1]: Big proponent of end-of-file-fixer

**Comment by polinaeterna** [comment]
@Skylion007 hi! I agree with @nateraw here, I'd better not force to use pre-commit so I'm not setting it up in the CI for now. And regarding end-of-file - currently it's being done by `black`.

**Pull request merged**